mvrx-navigation/build.gradle

```diff
@@ -0,0 +1,135 @@
+apply plugin: 'com.android.library'
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-android-extensions'
+apply plugin: 'jacoco'
+apply plugin: 'com.vanniktech.maven.publish'
+
+android {
+    compileSdkVersion versions.compileSdk
+    buildToolsVersion versions.buildTools
+    resourcePrefix "mvrx_"
+
+    defaultConfig {
+        minSdkVersion versions.minSdk
+        targetSdkVersion versions.targetSdk
+        consumerProguardFiles "proguard-rules.pro"
+    }
+
+    lintOptions {
+        abortOnError true
+        textReport true
+        textOutput "stdout"
+    }
+
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+
+    kotlinOptions {
+        jvmTarget = JavaVersion.VERSION_1_8.toString()
+    }
+
+    sourceSets {
+        main.java.srcDirs += 'src/main/kotlin'
+        test.java.srcDirs += 'src/test/kotlin'
+        test.res.srcDirs += 'src/test/res'
+    }
+
+    testOptions {
+        unitTests {
+            includeAndroidResources = true
+            all {
+                jacoco {
+                    includeNoLocationClasses = true
+                }
+            }
+        }
+    }
+}
+
+androidExtensions {
+    experimental = true
+}
+
+jacoco {
+    toolVersion = "0.8.2"
+}
+
+dependencies {
+    implementation project(":mvrx")
+    implementation project(":testing")
+
+    implementation libraries.lifecycleCommonJava8
+    implementation libraries.kotlin
+
+    testImplementation testLibraries.mockito
+    testImplementation testLibraries.junit
+    testImplementation testLibraries.roboeletric
+    testImplementation libraries.rxAndroid
+}
+
+// module specific dependencies
+dependencies {
```

Comment:
I am using your branch in a personal project of mine (that is the whole reason why I was pushing the refactors 😉).

@marukami I was trying to figure out why you made two separated segments of `dependencies` into this `build.gradle`.

Reply:
Some libraries already are saved at `dependencies.gradle`:

```Groovy
dependencies {
    implementation libraries.navigationFragment

    debugImplementation libraries.fragment
    debugImplementation libraries.fragmentKtx
    debugImplementation libraries.fragmentTesting

    testImplementation testLibraries.navigationTesting
}
```

Reply:
> I was trying to figure out why you made two separate segments of dependencies into this build.gradle. 

We do not need `dependencies.gradle` here as these libraries are not shared with any other module and having a separator like a 2nd dependencies seemed like a nice way to segment the core project dependencies from the module-specific dependencies.

> Does not api already suggests that a package is restricted to the module?

That rule seems a bit oversimplified. It's an OK rule of thumb inside a project module as for a Library I'm not sure. Tho in a library I don't think it's a restricted module otherwise then we would API include most things.

The assumption is here is the same as with `implementation` for Koltin STD and AppCompat.  In that the consumer is assumed to already be including the `navigation-component` modules otherwise they can't have a navigation graph to attach the `ViewModel`.

Tho, I've not managed many libraries and I migth be missing something here.